R/get_dayflow.R

```diff
@@ -44,8 +44,26 @@ get_dayflow <- function(){
 
     # read in the data
 
+    # set the column type to 'character'
     col_types <- readr::cols(.default = readr::col_character())
-    dat <- lapply(urls, readr::read_csv, col_types=col_types, show_col_types = FALSE, progress = FALSE)
+    # make an empty list
+    dat <- list()
+    # for every url...
+    for (i in 1:length(urls)){
+        # if the url corresponds to the dayflowcalculations2019 csv file,
+        if (grepl(pattern = "dayflowcalculations2019", x = urls[i])){
+            # read in the csv with some special parsing (read in the columns from 'Year' to 'X2' for the first 365 lines)
+            suppressWarnings(dat_file <- readr::read_csv(urls[i], n_max = 365, col_select = c(Year:X2),
```

Comment:
maybe the reason that this doesn't work is because the parser doesn't yet know about the column names? I'm not sure when the empty column at the end would get assigned "X2" but I can imagine how it might not recognize that as a column since it isn't listed in the file and R just gives that name later. If you want you could try selecting by index there instead. Normally I wouldn't advocate for that but we are selecting by index with n_max anyway, and I wouldn't expect this file to change

Reply:
Curiously, there was an empty column after "X2", and *that* was the one causing issues. For example, this would happen when I try previewing the file in Google Drive 🔍 
<img width="1421" alt="Screen Shot 2024-02-14 at 2 13 12 PM" src="https://github.com/goertler/inundation/assets/46659657/c3b8d762-69a0-4956-8aea-8bf2e8872bce">

This weird 30th column is completely blank yet R and Google Drive can somehow sense that it's there. 

I tried selecting by index instead of by name just now with `col_select = 1:29` and I got the first 29 columns but the same warning persists so I'm wondering if it's possible that `col_select` is applied after the csv is read in (?). 

Nonetheless, I think we're getting the correct columns/rows in the end so I suppressed the warnings to not cause concern 🤔

Reply:
yeah could also be that there is a bug in the readr package. seems okay to me to suppress the warning on just this file read, again since it isn't going to change. I'll go ahead and merge. thanks for doing this!